src/playlist-plugin.js

```diff
@@ -0,0 +1,271 @@
+   */
+  loadLastItem() {
+    const lastIndex = this.playlist_.getLastIndex();
+
+    return this.loadPlaylistItem(lastIndex);
+  }
+
+  /**
+   * Loads the next item in the playlist.
+   *
+   * @param {boolean} options.loadPoster - Whether or not the next item's poster image should be loaded
+   * @return {boolean} True if the next item was loaded successfully, false otherwise.
+   */
+  loadNextItem({ loadPoster = true } = {}) {
+    const nextIndex = this.playlist_.getNextIndex();
+
+    if (nextIndex === -1) {
+      return false;
+    }
+
+    return this.loadPlaylistItem(nextIndex, { loadPoster });
+  }
+
+  /**
+   * Loads the previous item in the playlist.
+   *
+   * @return {boolean} True if the previous item was loaded successfully, false otherwise.
+   */
+  loadPreviousItem() {
+    const previousIndex = this.playlist_.getPreviousIndex();
+
+    if (previousIndex === -1) {
+      return false;
+    }
+
+    return this.loadPlaylistItem(previousIndex);
+  }
+
+  /**
+   * Loads a specific playlist item.
+   *
+   * @param {Object} item - The playlist item to load.
+   * @param {boolean} options.loadPoster - Whether or not the item's poster image should be loaded
+   * @private
+   */
+  loadItem_(item, { loadPoster = true } = {}) {
+    this.player.trigger('beforeplaylistitem', item);
+
+    // Remove any textTracks from a previous item
+    this.clearExistingItemTextTracks_(this.player);
+
+    this.player.poster(loadPoster ? item.poster : '');
+    this.player.src(item.sources);
+
+    this.player.ready(() => {
+      this.addItemTextTracks_(item);
+
+      this.player.trigger('playlistitem', item);
+    });
+  }
+
+  /**
+   * Sets up event forwarding from the playlist to the player.
+   *
+   * @private
+   */
+  setupEventForwarding_() {
+    const playlistEvents = ['playlistchange', 'playlistadd', 'playlistremove', 'playlistsorted'];
+
+    playlistEvents.forEach((eventType) => this.playlist_.on(eventType, (event) => this.player.trigger(event)));
+  }
+
+  /**
+   * Plays the next item in the playlist.
+   *
+   * @private
+   */
+  playNext_() {
+    const loadedNext = this.loadNextItem({ loadPoster: false });
+
+    if (loadedNext) {
+      this.player.play();
+    }
+  }
+
+  /**
+   * Clears text tracks of the currently loaded item.
+   *
+   * @private
+   */
+  clearExistingItemTextTracks_() {
+    const textTracks = this.player.remoteTextTracks();
+    let i = textTracks && textTracks.length || 0;
+
+    // This uses a `while` loop rather than `forEach` because the
+    // `TextTrackList` object is a live DOM list (not an array).
+    while (i--) {
+      this.player.removeRemoteTextTrack(textTracks[i]);
+    }
+  }
+
+  /**
+   * Adds text tracks for a playlist item.
+   *
+   * @param {Object} item - The playlist item.
+   * @private
+   */
+  addItemTextTracks_(item) {
+    item.textTracks.forEach(this.player.addRemoteTextTrack.bind(this.player));
```

Comment:
This will only ever be passed sanitized playlist items, which have a `textTracks` property that defaults to an empty array.

Reply:
However, `sanitizePlaylistItem_()` does not do any type validation of `textTracks` or `poster` values that are passed in. So if an integrator passed in an unsupported `textTracks` value that is not an array, this would throw.